homeassistant/components/tod/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for Times of the Day integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any, cast
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_AFTER,
+    CONF_BEFORE,
+    CONF_NAME,
+    SUN_EVENT_SUNRISE,
+    SUN_EVENT_SUNSET,
+)
+from homeassistant.helpers import selector
+from homeassistant.helpers.helper_config_entry_flow import (
+    HelperConfigFlowHandler,
+    HelperFlowStep,
+)
+
+from .const import (
+    CONF_ABSOLUTE_TIME,
+    CONF_AFTER_OFFSET,
+    CONF_AFTER_TIME,
+    CONF_BEFORE_OFFSET,
+    CONF_BEFORE_TIME,
+    DOMAIN,
+)
+
+_BEFORE_AFTER_OPTIONS = [SUN_EVENT_SUNRISE, SUN_EVENT_SUNSET, CONF_ABSOLUTE_TIME]
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_AFTER): selector.selector(
+            {"select": {"options": _BEFORE_AFTER_OPTIONS}}
+        ),
+        vol.Optional(CONF_AFTER_TIME): selector.selector({"time": {"as_str": True}}),
+        vol.Required(CONF_AFTER_OFFSET, default={"seconds": 0}): selector.selector(
+            {"duration": {"as_dict": True}}
+        ),
+        vol.Required(CONF_BEFORE): selector.selector(
+            {"select": {"options": _BEFORE_AFTER_OPTIONS}}
+        ),
+        vol.Optional(CONF_BEFORE_TIME): selector.selector({"time": {"as_str": True}}),
+        vol.Required(CONF_BEFORE_OFFSET, default={"seconds": 0}): selector.selector(
+            {"duration": {"as_dict": True}}
+        ),
```

Comment:
The UX is not good, we should conditionally hide the time and duration selectors.
Maybe a version of the select selector which has: `value: {label: str, sub-selectors: [selector]}` items.

Reply:
Yeah... I'm not really sure if this is "presentable" in the current state :s
Don't get time wrong, I get this is the best possible option at this point... but...
It is highly confusing:

<img width="599" alt="image" src="https://user-images.githubusercontent.com/195327/158870767-a6be18b2-45cf-4100-b500-7429e37f08cb.png">